# HG changeset patch
# User Rik <rik@octave.org>
# Date 1422167232 28800
#      Sat Jan 24 22:27:12 2015 -0800
# Node ID e68267373191cbe4b54b9b30c7ca0b19b8faa639
# Parent  6928682a90b2f4b868fdd3adf668ca06e98f8b2d
chop.m: Calculate deflate as 1./inflate (bug #43734).

chop.m: Calculate deflate as 1./inflate (bug #43734).

diff --git a/scripts/general/chop.m b/scripts/general/chop.m
--- a/scripts/general/chop.m
+++ b/scripts/general/chop.m
@@ -50,17 +50,17 @@ function retval = chop (x, ndigits, base
     ## produces an integer that contains the digits we want to keep.
     ## Multiplying by deflate puts the decimal back where it belngs.
     ##
     ## Further scaling and rounding with the base factor produces a
     ## value with ndigits exactly divisible by base.  We skip that step
     ## unless base was explicitly provided.
 
     inflate = 10 .^ (ndigits - tmp);
-    deflate = 10 .^ (tmp - ndigits);
+    deflate = 1 ./ inflate;
     if (nargin == 2)
       retval = deflate .* round (x .* inflate);
     else
       retval = base .* deflate .* round (round (x .* inflate) ./ base);
     endif
   else
     print_usage ();
   endif
